minio/minio
# Fix SSEC multipart checksum replication

## Description

* Multipart SSEC checksums were not transferred.
* Remove key mismatch logging. This key is user controlled with SSEC.
* If source is SSEC and destination reports ErrSSEEncryptedObject, assume replication is good.

## How to test this PR?

Upload multipart with SSE-C and checksum (used a modified mc)

Verify that the replica has checksum. Again, modified mc:

```
mc: <DEBUG> HEAD /testbucket/TS40.txt?versionId=636377dc-feb4-4970-804d-46cc907f16e8 HTTP/1.1
Host: 127.0.0.1:9000
User-Agent: MinIO (windows; amd64) minio-go/v7.0.70 mc/DEVELOPMENT.GOGET
Authorization: AWS4-HMAC-SHA256 Credential=minio/20240611/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-checksum-mode;x-amz-content-sha256;x-amz-date;x-amz-server-side-encryption-customer-algorithm;x-amz-server-side-encryption-customer-key;x-amz-server-side-encryption-customer-key-md5, Signature=**REDACTED**
X-Amz-Checksum-Mode: ENABLED
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20240611T171617Z
X-Amz-Server-Side-Encryption-Customer-Algorithm: AES256
X-Amz-Server-Side-Encryption-Customer-Key: 0123456789012345678901234567890123456789010=
X-Amz-Server-Side-Encryption-Customer-Key-Md5: S1cFfu8lZZ98RPbWGV7KwA==

mc: <DEBUG> HTTP/1.1 200 OK
Content-Length: 400000000
Accept-Ranges: bytes
Content-Type: text/plain
Date: Tue, 11 Jun 2024 17:16:17 GMT
Etag: "2f8129e946cc9e960dd5c71e233db75d-24"
Last-Modified: Tue, 11 Jun 2024 17:03:31 GMT
Server: MinIO
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Origin
Vary: Accept-Encoding
X-Amz-Checksum-Crc32c: nZB0eg==-24
X-Amz-Id-2: dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8
X-Amz-Replication-Status: REPLICA
X-Amz-Request-Id: 17D802A7E2839574
X-Amz-Server-Side-Encryption-Customer-Algorithm: AES256
X-Amz-Server-Side-Encryption-Customer-Key-Md5: S1cFfu8lZZ98RPbWGV7KwA==
X-Amz-Version-Id: 636377dc-feb4-4970-804d-46cc907f16e8
X-Content-Type-Options: nosniff
X-Xss-Protection: 1; mode=block
....
```

Note replica has `X-Amz-Checksum-Crc32c` as expected and it is decoded correctly with provided SSEC key.

## Types of changes
